src/cargo/ops/cargo_compile/mod.rs

```diff
@@ -188,6 +188,7 @@ pub fn print<'a>(
         }
         let target_info = TargetInfo::new(gctx, &build_config.requested_kinds, &rustc, *kind)?;
         let mut process = rustc.process();
+        apply_env_config(gctx, &mut process)?;
```

Comment:
> (can add a test if desired, just tell me where)

See https://github.com/rust-lang/cargo/blob/master/tests/testsuite/rustc.rs

Looks like we have `--print` tests in there.  Unsure about custom targets.

We generally ask that PRs are split into
- One commit that adds the test, showing the currently broken behavior (ie it passes)
- The fix commit that updates the test to pass with the change

This makes the diff between the two commits show the change in behavior